sycl/CMakeLists.txt

```diff
@@ -113,16 +113,31 @@ if( NOT OpenCL_LIBRARIES )
     GIT_TAG           origin/master
     SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/OpenCL/icd"
     BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/icd_build"
+    INSTALL_DIR       "${CMAKE_CURRENT_BINARY_DIR}/icd_install"
     CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
                -DOPENCL_ICD_LOADER_HEADERS_DIR=${OpenCL_INCLUDE_DIRS}
-               -DCMAKE_INSTALL_PREFIX=${LLVM_BINARY_DIR}
+               -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
                -DCMAKE_INSTALL_LIBDIR:PATH=lib${LLVM_LIBDIR_SUFFIX}
                ${AUX_CMAKE_FLAGS}
     STEP_TARGETS      configure,build,install
     DEPENDS           ocl-headers
     BUILD_BYPRODUCTS ${OpenCL_LIBRARIES}
   )
+  ExternalProject_Add_Step(ocl-icd llvminstall
+    COMMAND ${CMAKE_COMMAND} -E copy_directory <INSTALL_DIR>/ ${LLVM_BINARY_DIR}
+    COMMENT "Installing ocl-icd into the LLVM binary directory"
+    DEPENDEES install
+  )
+  # Optionally install the ICD library along with LLVM.
+  option(OpenCL_INSTALL_KHRONOS_ICD_LOADER
+    "Install the Khronos ICD Loader built as part of LLVM" OFF)
+  if (OpenCL_INSTALL_KHRONOS_ICD_LOADER)
+    install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/icd_install/"
+      DESTINATION "."
+      COMPONENT opencl-icd
+    )
+  endif()
```

Comment:
Am I right this only adds ICD installation to `install` target, and not the `deploy-sycl-toolchain`?

Reply:
You are absolutely right. That was an oversight of me. :frowning:

Now I added the installation of the library to the `deploy-sycl-toolchain` target as well, in case `OpenCL_INSTALL_KHRONOS_ICD_LOADER` is set to `TRUE`.